### PR TITLE
bazel*: build and test frontend code

### PR DESCRIPTION
**Pull request by squat**
Implements: INST-744

This commit adds fully hermetic builds of the JavaScript and Go bindata.
Bazel will now manage its own copy of the JavaScript node_modules
directory and use it for all builds.

To build the JavaScript, run
```sh
bazel build installer/frontend
```

To build the bindata, run
```sh
bazel build installer:bindata
```

To run the frontend tests, run:
```sh
bazel test installer/frontend:frontend_test
```

To run the intsaller GUI tests, export the required environtment
variables as specified in `installer/frontend/ui-tests/README.md` and run:
```sh
bazel test installer/frontend:baremetal_installer_gui_test --test_env=TF_VAR_tectonic_license_path --test_env=TF_VAR_tectonic_pull_secret_path
```

cc @alexsomesan

### CODE REVIEW

**Comment by squat** [comment]
ok to test

File: installer/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+)
+
+genrule(
+    name = "bindata",
+    srcs = [
+        ":asset_files",
+        "//installer/frontend:frontend",
+    ],
+    outs = ["assets/bindata.go"],
+    cmd = "mkdir -p $(@D)/frontend/scripts && cp $(location //installer/frontend:frontend) $(@D)/frontend/scripts && cp -r installer/assets/* $(@D) && $(location //installer/vendor/github.com/jteeuwen/go-bindata/go-bindata:go-bindata) -o $@ -pkg assets -prefix $(@D) -ignore=$@ -ignore=doc.go -ignore=assets.go $(@D)/...",
```
Comment (alexsomesan) [inline installer/BUILD.bazel]: This is the stuff that makes "Make" brittle.
On a more immediate note, it fails on macOS due to divergent flags to `cp`.

Here's what I get:
```
bazel build installer/frontend
WARNING: /Users/alex/go/src/github.com/coreos/tectonic-installer/installer/frontend/BUILD.bazel:20:1: target 'node_modules' is both a rule and a file; please choose another name for the rule
INFO: Analysed target //installer/frontend:frontend (15 packages loaded).
INFO: Found 1 target...
ERROR: /Users/alex/go/src/github.com/coreos/tectonic-installer/installer/frontend/BUILD.bazel:20:1: Installing JavaScript dependencies... //installer/frontend:node_modules failed (Exit 1)
cp: the -H, -L, and -P options may not be specified with the -r option.
Target //installer/frontend:frontend failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 14.637s, Critical Path: 0.18s
FAILED: Build did NOT complete successfully
```

Comment (squat) [reply installer/BUILD.bazel]: @alexsomesan please try again now. I forgot that `-r` is not defined as a flag for `cp` for MacOS. Instead, we can use `cp -RL`. This is easier and cleaner than resorting to a tar pipe IMO.

Comment (alexsomesan) [reply installer/BUILD.bazel]: Confirmed working! Thanks for fixing it.

We should be good with this for now, but ideally we do all of this in SkyLark rules as a follow-up.

**Pull request merged**